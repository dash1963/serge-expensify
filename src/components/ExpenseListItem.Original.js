
import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { removeExpense } from '../actions/expenses';


const ExpenseListItem = ( {dispatch, id, description, amount, createdAt} = {}) => (
    <div className="container">
       <table className="table table-striped">
          <tbody>
         <tr>
            <td><Link to={`/edit/${id}`}>{description}</Link></td>
            <td>{amount}</td>
            <td>{createdAt}</td>
            <td><button onClick={() => {
                        dispatch(removeExpense({id}));  
                    }}>remove
                </button></td>
          </tr>
          </tbody>
       </table>              


    </div>
);


const mapStateToProps = (state) => {
    return {
         expenses: state.expenses
    };
}
export default connect()(ExpenseListItem);



