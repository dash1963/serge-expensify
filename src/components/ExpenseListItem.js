
import React from 'react';
import { Link } from 'react-router-dom';


const ExpenseListItem = ( {dispatch, id, description, amount, createdAt} = {}) => (
    <div className="container">
       <table className="table table-striped">
          <tbody>
         <tr>
            <td><Link to={`/edit/${id}`}>{description}</Link></td>
            <td>{amount}</td>
            <td>{createdAt}</td>
          </tr>
          </tbody>
       </table>              


    </div>
);

export default ExpenseListItem;



