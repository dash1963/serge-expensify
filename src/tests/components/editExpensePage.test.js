import React from 'react';
import {shallow } from 'enzyme';
import expenses from '../fixtures/expenses';
import { editExpensePage } from '../../components/EditExpensePage';

let editExpense , removeExpense, history, wrapper;

beforeEach(() => {
    editExpense = jest.fn();
    removeExpense = jest.fn();
    history = { push: jest.fn()};
    wrapper = shallow (
        <editExpensePage 
           editExpense   = {editExpense}
           removeExpense = {removeExpense}
           history       = {history}
           expense       = {expenses[0]}
        />
    );
});

test('Should render editExpensePage', () => {
    expect(wrapper).toMatchSnapshot();
});

test('Should handle edit expense', () => {
  wrapper.find('ExpenseForm').prop('onSubmit')(expenses[0]);

   expect(history.push).toHaveBeenLastCalledWith('/');
   expect(editExpense).toHaveBeenLastCalledWith(
          expenses[0].id , expenses[0]
         );
});

test('Shound handle remove expense', () => {
   wrapper.find('button').simulate('click');

   expect(history.push).toHaveBeenLastCalledWith('/');

   expect(removeExpense).toHaveBeenLastCalledWith({
          id: expenses[0].id});
});
