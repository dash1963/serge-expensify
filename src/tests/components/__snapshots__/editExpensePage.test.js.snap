// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render editExpensePage 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <editExpensePage
    editExpense={[Function]}
    expense={
        Object {
            "amount": 195,
            "createdAt": 0,
            "description": "Gum",
            "id": "1",
            "note": "",
          }
    }
    history={
        Object {
            "push": [Function],
          }
    }
    removeExpense={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "editExpense": [Function],
      "expense": Object {
        "amount": 195,
        "createdAt": 0,
        "description": "Gum",
        "id": "1",
        "note": "",
      },
      "history": Object {
        "push": [Function],
      },
      "removeExpense": [Function],
    },
    "ref": null,
    "rendered": null,
    "type": "editExpensePage",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "editExpense": [Function],
        "expense": Object {
          "amount": 195,
          "createdAt": 0,
          "description": "Gum",
          "id": "1",
          "note": "",
        },
        "history": Object {
          "push": [Function],
        },
        "removeExpense": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": "editExpensePage",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
